source/common/router/config_impl.h

```diff
@@ -313,6 +343,7 @@ class RouteEntryImplBase : public RouteEntry,
   // Default timeout is 15s if nothing is specified in the route config.
   static const uint64_t DEFAULT_ROUTE_TIMEOUT_MS = 15000;
 
+  CorsPolicySharedPtr cors_policy_;
```

Comment:
per other comments, I think if we remove shared_ptr, this can become unique_ptr

Reply:
With unique pointer do you mean the `CorsPolicy*` you wrote above or `unique_ptr`?

Reply:
I just meant that internally you can store as unique_ptr<CorsPolicy) then just return cors_policy_.get()

Reply:
Understood. If using `new` then use unique_ptr. But since I will just initialize it in the constructor I don't need `new` so I will just make it a member like `RetryPolicy` etc. 
( https://github.com/lyft/envoy/blob/master/source/common/router/config_impl.h#L327-L333 )